packages/components/molecules/f-restaurant-card/src/components/subcomponents/_tests/IconText.test.js

```diff
@@ -0,0 +1,88 @@
+import { shallowMount } from '@vue/test-utils';
+import sut from '../IconText.vue';
+
+describe('IconText.vue', () => {
+    it('displays the provided text', () => {
+        // arrange
+        const propsData = {
+            text: 'foo'
+        };
+
+        // act
+        const wrapper = shallowMount(sut, { propsData });
+
+        const visibleText = wrapper.find('[data-test-id="icon-text-visible-text"]').text();
+
+        // assert
+        expect(visibleText).toBe('foo');
+    });
+
+    it('displays the provided accessible Text', () => {
+        // arrange
+        const propsData = {
+            text: 'foo',
+            accessibleText: 'bar'
+        };
+
+        // act
+        const wrapper = shallowMount(sut, { propsData });
+
+        const accessibleText = wrapper.find('[data-test-id="icon-text-accessible-text"]').text();
+
+        // assert
+        expect(accessibleText).toBe('bar');
+    });
+
+    it('fallbacks to text if accessible text is missing', () => {
+        // arrange
+        const propsData = {
+            text: 'foo'
+        };
+
+        // act
+        const wrapper = shallowMount(sut, { propsData });
+
+        const accessibleText = wrapper.find('[data-test-id="icon-text-accessible-text"]').text();
+
+        // assert
+        expect(accessibleText).toBe('foo');
+    });
+
```

Comment:
again, I question the purpose of duplicating the default text if no accessible string provided - if anything that might make the screen reader repeat the same thing twice for each restaurant. As I said in my other comment this might need to be something we add in future due to translations etc.

Reply:
it won't repeat it twice. The block that has `aria-hidden="true"` will be ignored by screen readers. Only the accessible text (or the fallback to regular text if the accessible text is not provided) will be visible to screen readers.